### ISSUE TITLE
Add an 'event' command

### ISSUE THREAD

**Issue by danielballan** [NSLS-II/bluesky#941]
Suggestion by @tacaswell which I am just recording:

Add an `'event'` command so that

```python
Msg('create')
Msg('read', obj1)
Msg('read', obj2)
Msg('save')
```

can be accomplished by

```python
Msg('event', [obj1, obj2])
```

There is really no reason one should need to interleave other commands between these; there is no situation we can think of where `'event'` is not sufficient. This is a nice simplification.

**Comment by danielballan** [comment]
Having just written that, I have just thought of one such situation where we need `'create'` / `'save'` -- the case where we want a plan to be able to see a reading before the Event is created and decide whether to `'save'` or `'drop'`. I therefore amend my previous statement: the proposed `'event'` command is a nice simplification for the common case, reducing the number of messages per plan,  but `'create'` and `'save'` still have an important use.

**Comment by prjemian** [comment]
Will a description document be generated that relates to this event document?  Will there be a 1-to-1 relationship for such description documents when an event is generated this way?

**Comment by danielballan** [comment]
Yes, the behavior of the RunEngine would be precisely identical in these two modes. It simplify removes the overhead of processing multiple Msgs.

**Comment by prjemian** [comment]
I think I understand.  Consider this scenario:

```
Msg('event', [obj1, obj2])
Msg('event', [obj1, obj2])
```

generates two descriptor documents, essentially duplicates but for UUID.

If so, can some smarts be added in RE to see if a descriptor has already been generated for the same signal list: `[obj1, obj2]`?

**Comment by danielballan** [comment]
Yes, in fact those smarts already exist:

```
Msg('create')
Msg('read', obj1)
Msg('read', obj2)
Msg('save')  # in the RunEngine._save coroutine, the RunEngine builds and emits a new Descriptor and then the Event
Msg('create')
Msg('read', obj1)
Msg('read', obj2)
Msg('save')  # in the RunEngine._save coroutine, the RunEngine finds that it already has emitted an appropriate Descriptor and merely emits the Event
```

I would apply the same logic to
```
Msg('event', [obj1, obj2])  # emit Descriptor, Event
Msg('event', [obj1, obj2])  # emit Event
```
